email/reporter.go

```diff
@@ -465,12 +468,15 @@ func (r *Reporter) openReport(entity database.NCMECReport) (uint64, error) {
 	}
 	if err != nil {
 		// update the email and set the report err
-		err = errors.Compose(err, r.staticAbuseDatabase.UpdateReportNoLock(entity, bson.M{
+		updateErr := r.staticAbuseDatabase.UpdateReportNoLock(entity, bson.M{
 			"$set": bson.M{
 				"filed_err": err.Error(),
 			},
-		}))
-		logger.Errorf("failed to open report, err '%v'", err)
+		})
+		if updateErr != nil {
+			logger.Errorf("failed to open report, err '%v'", updateErr)
+			err = errors.Compose(err, updateErr)
+		}
```

Comment:
This update is different from the original behaviour. The previous version of the code logged this message regardless of the success of `UpdateReportNoLock` and logged the composed error. Are you sure we don't want to do that anymore?

Reply:
thanks, oversight